src/applications/disability-benefits/526EZ/actions/index.js

```diff
@@ -1,64 +1,138 @@
+  return {
+    type: RESET_PRESTART_STATUS
+  };
+}
 
-    const delay = (ms) => new Promise((resolve, reject) => {
-      ITFStatus = ITFStatuses.active;
-      onChange({ ITFStatus });
+export function resetPrestartDisplay() {
+  return {
+    type: RESET_PRESTART_DISPLAY
+  };
+}
 
-      // TODO: if the backend handles resubmission, this check can be removed
-      if (ITFStatus === ITFStatuses.active) {
-        setTimeout(resolve, ms);
+export function checkITFRequest(dispatch, hasSavedForm) {
+  return apiRequest(
+    '/intent_to_file',
+    null,
+    ({ data }) => {
+      let status;
+      let expirationDate;
+      const itfList = data.attributes.intentToFile;
+      if (!itfList || (Array.isArray(itfList) && itfList.length === 0)) {
+        status = PRESTART_STATUSES.none;
       } else {
-        const errorMessage = 'Network request failed';
-        onChange({
-          ITFStatus,
-          errorMessage
-        });
-        reject(errorMessage);
+        const activeList = itfList.filter(itf => itf.status === 'active');
+        if (activeList.length === 0) {
+          status = PRESTART_STATUSES.expired;
+          expirationDate = itfList.sort((a, b) => a.expirationDate - b.expirationDate)[0].expirationDate;
+        } else {
+          status = PRESTART_STATUSES.retrieved;
+          expirationDate = activeList[0].expirationDate;
+        }
       }
-    });
+      dispatch(setPrestartStatus(status, expirationDate));
+      return status;
+    },
+    ({ errors }) => {
+      const errorMessage = 'Network request failed';
+      const status = hasSavedForm ? PRESTART_STATUSES.notRetrievedSaved : PRESTART_STATUSES.notRetrievedNew;
+      Raven.captureMessage(`vets_itf_error: ${errorMessage}`);
+      dispatch(setPrestartStatus(status, errors));
+      return status;
+    }
+  );
+}
 
-    return delay(2000);
-  };
+export function submitITFRequest(dispatch, successStatus, errorStatus) {
+
+  return apiRequest(
+    '/intent_to_file/compensation',
+    { method: 'POST' },
+    ({ data }) => {
+      dispatch(setPrestartStatus(successStatus, data.attributes.intentToFile.expirationDate));
+      return successStatus;
+    },
+    ({ errors }) => {
+      const errorMessage = 'Network request failed';
+      Raven.captureMessage(`vets_itf_error: ${errorMessage}`);
+      dispatch(setPrestartStatus(errorStatus, errors));
+      return errorStatus;
+    }
+  );
 }
 
-//   apiRequest(
-//     `${intentToFileUrl}`,
-//     null,
-//     ({ data }) => {
-//       const ITFStatus = data.attributes.ITFStatus
-//       onChange({ ITFStatus });
-//       return ITFStatus === 'active';
-//     },
-//     ({ errors }) => {
-//       const errorMessage = 'Network request failed';
-//       onChange({
-//         ITFStatus,
-//         errorMessage
-//       });
-//       Raven.captureMessage(`vets_itf_error: ${errorMessage}`);
-//     }
-//   );
-// };
+// function fakeITFRequest(x, cb) { // remove once local testing complete
```

Comment:
This can probably be removed now, right?

Reply:
These are for local testing/review purposes, they can be removed.